src/rust/ru.desu/src/helper.rs

```diff
@@ -0,0 +1,134 @@
+use crate::constants;
+use aidoku::std::{ValueRef};
+use aidoku::{
+	error::{Result},
+	helpers::uri::encode_uri,
+	prelude::*,
+	std::net::{HttpMethod, Request},
+	Filter, FilterType, Manga, MangaPageResult};
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+use crate::parser::debug;
+
+pub fn get_manga_url(id: &str) -> String {
+	format!("{}/{}", constants::BASE_API_URL, encode_uri(id))
+}
+
+pub fn get_chapter_url(manga_id: &str, chapter_id: &str) -> String {
+	format!("{}/chapter/{}", get_manga_url(manga_id), encode_uri(chapter_id))
+}
+
+pub fn fetch_json<T: AsRef<str>>(url: T, fetch_for: &str) -> Result<ValueRef>
+{
+	debug!("Fetching for {}, fetching url {}", fetch_for, url.as_ref());
+
+	let request = Request::new(url, HttpMethod::Get)
+		.header("User-Agent", constants::USER_AGENT)
+		.header("Referer", constants::BASE_URL);
+
+	Ok(request.json()
+		.expect(format!("Failed to find JSON for {}", fetch_for).as_str())
+		.as_object()?
+		.get("response"))
+}
+
+pub fn get_search_url(filters: Vec<Filter>, page: i32) -> String
+{
+	debug!("Generating search");
+
+	let mut url = String::from(constants::BASE_API_URL);
+	let mut query: Vec<String> = Vec::new();
+	let mut genres: Vec<String> = Vec::new();
+	let mut kinds: Vec<String> = Vec::new();
+	let mut statuses: Vec<String> = Vec::new();
+	let mut order = String::new();
+
+	if page > 1 {
+		query.push(format!("page={}", page.to_string()));
+	}
+	query.push(format!("limit={}", constants::PAGE_LIMIT));
+
+	for filter in filters {
+		debug!("Filter: {}", filter.name);
+		match filter.kind {
+			FilterType::Title => {
+				if let Ok(filter_value) = filter.value.as_string() {
+					query.push(format!("search={}", encode_uri(filter_value.read())));
+				}
+			},
+			FilterType::Genre => {
+				if let Ok(obj_id) = filter.object.get("id").as_string() {
+					let id = obj_id.read();
+					match filter.value.as_int().unwrap_or(-1) {
+						0 => genres.push(format!("!{}", id)),
+						1 => genres.push(id),
+						_ => continue
+					}
+				}
+			},
+			FilterType::Check => {
+				if let Ok(obj_id) = filter.object.get("id").as_string() {
+					let id = obj_id.read();
+					let parts: Vec<&str> = id.split('|').collect();
```

Comment:
`split_once` is better, but i think what i said on discord would be better. you don't need to split if you just assume the first character of the id will always be the kind, and the rest will be the real id. you're doing lots of unnecessary work here by splitting even though you know the ids are all the same format